### PR TITLE
[evm] check null pointer before dump calldata

### PR DESCRIPTION
**Pull request by dustinxie**
this might be the reason for api node restart

### CODE REVIEW

**Review by Liuhaai — APPROVED** [review]

File: action/protocol/execution/evm/evm.go [hunk 1]
```diff
@@ -439,7 +446,6 @@ func evmErrToErrStatusCode(evmErr error, g genesis.Blockchain, height uint64) (e
 			case "no compatible interpreter":
 				errStatusCode = uint64(iotextypes.ReceiptStatus_ErrNoCompatibleInterpreter)
 			default:
-				log.L().Error("evm internal error", zap.Error(evmErr))
```
Comment (Liuhaai) [inline action/protocol/execution/evm/evm.go:442]: why is this log removed

File: action/protocol/execution/evm/evm.go [hunk 2]
```diff
@@ -371,8 +372,14 @@ func executeInEVM(evmParams *Params, stateDB *StateDBAdapter, g genesis.Blockcha
 	if evmErr != nil {
 		errCode = evmErrToErrStatusCode(evmErr, g, blockHeight)
 		if errCode == uint64(iotextypes.ReceiptStatus_ErrUnknown) {
+			var address string
+			if evmParams.contract != nil {
```
Comment (Liuhaai) [inline action/protocol/execution/evm/evm.go:376]: when contract address is empty (""), will the contract field be nil?

File: pkg/log/fields.go [hunk 3]
```diff
@@ -20,5 +20,5 @@ import (
 
 // Hex creates a zap field which convert binary to hex.
 func Hex(k string, d []byte) zap.Field {
-	return zap.String(k, hex.EncodeToString(d))
+	return zap.String(k, "0x"+hex.EncodeToString(d))
```
Comment (CoderZhi) [inline pkg/log/fields.go:23]: is this related to the bug? if not, revert it

File: action/protocol/execution/evm/evm.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/execution/evm/evm.go:442]: it becomes the dump data in the PR

**Review by CoderZhi — APPROVED** [review]

File: action/protocol/execution/evm/evm.go (hunk 2 above)
Comment (dustinxie) [reply action/protocol/execution/evm/evm.go:376]: yes, empty "" --> evmParams.contract == nil
